databricks_cli/cluster_policies/api.py

```diff
@@ -20,18 +20,42 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import json
+
 from databricks_cli.sdk import PolicyService
 
 
 class ClusterPolicyApi(object):
     def __init__(self, api_client):
         self.client = PolicyService(api_client)
 
-    def create_cluster_policy(self, json):
-        return self.client.client.perform_query('POST', '/policies/clusters/create', data=json)
+    @staticmethod
+    def convert_to_json_string(data):
+        if isinstance(data, str):
+            return data
+        if isinstance(data, dict):
+            return json.dumps(data)
+        raise TypeError(
+            "Only str or dict type are accepted, but got: {}.".format(type(data))
+        )
+
+    @staticmethod
+    def format_policy_for_api(policy):
+        policy["definition"] = ClusterPolicyApi.convert_to_json_string(
+            policy["definition"]
+        )
```

Comment:
Thx for the review, I merged the two functions into one.
BTW, it seems that the `definition` key is indeed always set as per this API doc, so I'm using `policy["definition"]` instead of `policy.get`.
* create: https://docs.databricks.com/dev-tools/api/latest/policies.html#operation/create-cluster-policy
* edit :https://docs.databricks.com/dev-tools/api/latest/policies.html#operation/edit-cluster-policy

Reply:
It has to be, but it might not be when user specified.

Reply:
If we use `policy.get("definition", None)`, and user forgets to add the key `definition`, he will get an error only after calling the Databricks API, it's a network consommation.

But if we use `policy["definition"]`, user will get an error locally right now.

IMHO, if the user forgets the key, both methods will return an error, but the later one is faster.

Reply:
@pietern 
Please correct me if my understanding is wrong.